### PR TITLE
Add GitHub Action to check profile filenames

### LINKED ISSUE #3692: [FEATURE] GitHub Action to check profile filename matches the GitHub username
### Description

We have a label `LGTM` added when the author of the profile creates/edits their profile and the file name is the exact match

But this can get missed, so it would be great to have a GitHub Action that fails if these checks are not matched...

- is the author of the PR the owner of the file, for example for my GitHub user `eddiejaoude`, my file would have to match exactly to `data/eddiejaoude.json` (case sensitive)

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by dan-mba**
## Fixes Issue

Closes #3692 

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

### CODE REVIEW

File: .github/scripts/check-filenames.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+const fileNames = process.env.FILENAMES.split(' ');
+const baseFileNames = process.env.BASE_FILENAMES.split(' ');
+const user = process.env.USERNAME;
+let exitCode = 0;
+
+// Remove files not changed in this PR from list
+if (baseFileNames.length > 0) {
+  for(const file of baseFileNames) {
+    const index = fileNames.indexOf(file);
```
Comment (eddiejaoude) [inline .github/scripts/check-filenames.js:9]: Would using a `.filter` make this easier to read?

**Review by eddiejaoude — DISMISSED** [review]
Looks good 👍 , just 1 small question

File: .github/scripts/check-filenames.js (hunk 1 above)
Comment (dan-mba) [reply .github/scripts/check-filenames.js:9]: Probably. I will work with it on my test repo & see how it goes.

**Review by iamthecloverly — APPROVED** [review]

**Review by iamthecloverly — APPROVED** [review]

**Comment by Pradumnasaraf** [comment]
Looks good! 🎉

**Review by eddiejaoude — APPROVED** [review]
Thanks 👍

**Pull request merged**